src/Infrastructure/Riganti.Utils.Infrastructure.Services/Facades/CrudFacadeBase.cs

```diff
@@ -201,9 +262,29 @@ protected virtual void PopulateDetailToEntity(TDetailDTO detail, TEntity entity)
         /// Validates that the entity detail can be displayed by the user. If the user does not have permissions, the method should throw an exception.
         /// </summary>
         /// <param name="entity"></param>
-        protected virtual Task ValidateReadPermissions(TEntity entity)
+        protected virtual Task ValidateReadPermissionsAsync(TEntity entity)
```

Comment:
To be consistent (async method after sync), move this method after `ValidateReadPermissions(TEntity entity)`.